ui/src/contexts/auth-context.tsx

```diff
@@ -0,0 +1,55 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+
+interface AuthContextType {
+  authEnabled: boolean | null
+  isAuthenticated: boolean | null
+  refreshAuthStatus: () => void
+}
+
+const AuthContext = createContext<AuthContextType>({
+  authEnabled: null,
+  isAuthenticated: null,
+  refreshAuthStatus: () => {},
+})
+
+export const AuthProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [authEnabled, setAuthEnabled] = useState<boolean | null>(null)
+  const [isAuthenticated, setIsAuthenticated] = useState<boolean | null>(null)
+
+  const fetchAuthStatus = async () => {
+    try {
+      const statusRes = await fetch('/api/authentication/status', {
+        credentials: 'include',
+      })
+      const authEnabledHeader = statusRes.headers.get('X-Auth-Enabled')
```

Comment:
I can't see a need for this X-Auth-Enabled header stuff. `/api/authentication/status` could simply require authentication, and if a 401 is returned then the provided credentials (if any) are incorrect. The user can then be redirected to the login page. 

You don't even need to do that to be honest if you just do this: https://github.com/jorenn92/Maintainerr/pull/1607/files#r1976728680

The UI will start loading in and we'll attempt to make API calls, and then the user will be promptly redirected to the login page.